azure-devops/create-prdiff.ps1

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+[CmdletBinding(PositionalBinding = $False)]
+Param(
+    [Parameter(Mandatory = $True)]
+    [String]$DiffFile
+)
+
+Start-Process -FilePath 'git' -ArgumentList 'diff' `
+    -NoNewWindow -Wait `
+    -RedirectStandardOutput $DiffFile
+if (0 -ne (Get-Item -LiteralPath $DiffFile).Length) {
+    $msg = @(
+        'The formatting of the files in the repo was not what we expected.'
+        'Please access the diff from format.diff in the build artifacts,'
+        'and apply it with `git apply`.'
+        'Alternatively, you can run the `format` CMake target:'
+        '    cmake --build <builddir> --target format'
+        ''
+    )
+    Write-Host ($msg -join "`n")
+    Write-Host [string](Get-Content -LiteralPath $DiffFile)
+    throw

```

Comment:
When clang-format fails (see #2681), the Checks tab simply says `PowerShell exited with code '1'.` and one has to click on "View more details on Azure Pipelines" to see the message being printed out there. The current behavior emits the diff on the Checks tab.

I believe that this is controlled by whether a script emits anything on the standard error stream - that's how `validate.exe` gets its messages displayed on the Checks tab, and how they are treated as failure.

I am a PowerShell novice, but would (1) replacing [`Write-Host`](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/write-host?view=powershell-7.2) with [`Write-Error`](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/write-error?view=powershell-7.2) and (2) omitting the `throw` be sufficient to get this output captured in the Checks tab, and for it to be detected as a failure without the ominous-sounding ScriptHalted RuntimeException?